src/tasks/LargeFileUploadTask.ts

```diff
@@ -259,7 +300,14 @@ export class LargeFileUploadTask {
 	 * @returns The promise resolves to cancelled response
 	 */
 	public async cancel(): Promise<any> {
-		return await this.client.api(this.uploadSession.url).delete();
+		const cancelResponse = await this.client
+			.api(this.uploadSession.url)
+			.responseType(ResponseType.RAW)
+			.delete();
+		if (cancelResponse.status === 204) {
```

Comment:
it'd be nice to document the cancel capability in the design doc, do you mind taking a stab at it please?

Reply:
Done